Mathlib/LinearAlgebra/RootSystem/Finite/Nondegenerate.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2024 Scott Carnahan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Carnahan
+-/
+import Mathlib.LinearAlgebra.BilinearForm.Basic
+import Mathlib.LinearAlgebra.Dimension.LinearMap
+import Mathlib.LinearAlgebra.Dimension.Localization
+import Mathlib.LinearAlgebra.RootSystem.Finite.CanonicalBilinear
+import Mathlib.LinearAlgebra.RootSystem.RootPositive
+
+/-!
+# Nondegeneracy of the polarization on a finite root pairing
+We show that if the base ring of a finite root pairing is linearly ordered, then the canonical
+bilinear form is root-positive and positive-definite on the span of roots.
+From these facts, it is easy to show that Coxeter weights in a finite root pairing are bounded
+above by 4.  Thus, the pairings of roots and coroots in a root pairing are restricted to the
+interval `[-4, 4]`.  Furthermore, a linearly independent pair of roots cannot have Coxeter weight 4.
+For the case of crystallographic root pairings, we are thus reduced to a finite set of possible
+options for each pair.
+Another application is to the faithfulness of the Weyl group action on roots, and finiteness of the
+Weyl group.
+## Main results:
+ * `rootForm_rootPositive`: `RootForm` is root-positive.
```

Comment:
We need to fully qualify names to make our docs links work
```suggestion
 * `RootPairing.rootForm_rootPositive`: `RootForm` is root-positive.
```
(likewise the two below)